distributed/nanny.py

```diff
@@ -437,9 +434,19 @@ async def instantiate(self) -> Status:
         return result
 
     @log_errors
-    async def plugin_add(self, plugin=None, name=None):
+    async def plugin_add(
+        self, plugin: NannyPlugin | bytes, name: str | None = None
+    ) -> dict[str, Any]:
         if isinstance(plugin, bytes):
             plugin = pickle.loads(plugin)
+        if not isinstance(plugin, NannyPlugin):
+            warnings.warn(
+                "Registering duck-typed plugins has been deprecated. "
```

Comment:
How would expect this to be covered by `test_duck_typed_nanny_plugin_is_deprecated` but coverage is saying otherwise; how comes?